File: src/FoodSharing.Site/Models/Announcements/AnnouncementShortInfo.cs
namespace FoodSharing.Site.Models.Announcements;

public class AnnouncementShortInfo
{
    public Guid Id { get; }
    public String Name { get; }
    public String Description { get; }
    public String CityName { get; }
    public String MainImgUrl { get; }
    public DateTime CreatedAt { get; }
    public Boolean IsFavorite { get; }

    public AnnouncementShortInfo(
       Announcement announcement, String cityName, Boolean isFavorite
    )
    {
        Id = announcement.Id;
        Name = announcement.Name;
        Description = announcement.Description;
        CityName = cityName;
        MainImgUrl = announcement.ImagesUrls[0];
        CreatedAt = announcement.CreatedAt;
        IsFavorite = isFavorite;
    }
}

File: src/FoodSharing.Site/ClientApp/src/domain/announcements/announcementShortInfo.ts

export class AnnouncementShortInfo {
    constructor(
        public id: string,
        public name: string,
        public description: string,
        public cityName: string,
        public mainImgUrl: string,
        public createdAt: Date,
        public isFavorite: boolean
    ) { }
}

export function mapToAnnouncementShortInfo(data: any) {
    return new AnnouncementShortInfo(
        data.id,
        data.name,
        data.description,
        data.cityName,
        data.mainImgUrl,
        data.createdAt,
        data.isFavorite
    )
}